diff --git b/opendbc/car/hyundai/carcontroller.py a/opendbc/car/hyundai/carcontroller.py
index 8fc563b1..02c8f64b 100644
--- b/opendbc/car/hyundai/carcontroller.py
+++ a/opendbc/car/hyundai/carcontroller.py
@@ -159,6 +159,7 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
 
     # For future parametrization / tuning
     self.angle_enable_smoothing_factor = True
+    self.angle_limit_speed_threshold = 32.0 / 3.6
 
     self._params = Params() if PARAMS_AVAILABLE else None
     if PARAMS_AVAILABLE:
@@ -174,6 +175,13 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
       self.params.ANGLE_TORQUE_OVERRIDE_CYCLES = int(self._params.get("HkgTuningOverridingCycles") or self.params.ANGLE_TORQUE_OVERRIDE_CYCLES)
       self.angle_enable_smoothing_factor = self._params.get_bool("EnableHkgTuningAngleSmoothingFactor")
 
+      speed_threshold = self._params.get("HkgTuningAngleCustomLimitMaxSpeedKph", return_default=True)
+      if speed_threshold is None:
+        speed_threshold_mps = 32.0 / 3.6
+      else:
+        speed_threshold_mps = float(speed_threshold) / 3.6
+      self.angle_limit_speed_threshold = max(0.0, speed_threshold_mps)
+
     self.angle_torque_reduction_gain_controller = TorqueReductionGainController(
       angle_threshold=.3,
       debounce_time=.1,
@@ -209,6 +217,12 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
       if self.angle_enable_smoothing_factor and abs(v_ego_raw) < CarControllerParams.SMOOTHING_ANGLE_MAX_VEGO:
         desired_angle = sp_smooth_angle(v_ego_raw, desired_angle, self.apply_angle_last)
 
+      max_lat_accel = self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL
+      max_lat_jerk = self.params.ANGLE_LIMITS.MAX_LATERAL_JERK
+      if (self.CP.carFingerprint == CAR.KIA_EV9) and (v_ego_raw <= self.angle_limit_speed_threshold):
+        self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL = 4.2
+        self.params.ANGLE_LIMITS.MAX_LATERAL_JERK = 4.2
+
       apply_angle = apply_steer_angle_limits_vm(desired_angle, self.apply_angle_last, v_ego_raw, CS.out.steeringAngleDeg, CC.latActive, self.params, self.VM)
 
       # Historically we applied the baseline Santa Fe model on top to match Panda's hardcoded baseline.
@@ -224,6 +238,9 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
         apply_angle = apply_steer_angle_limits_vm(apply_angle or desired_angle, self.apply_angle_last, v_ego_raw, CS.out.steeringAngleDeg, CC.latActive,
                                                   self.params, self.BASELINE_VM)
 
+      self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL = max_lat_accel
+      self.params.ANGLE_LIMITS.MAX_LATERAL_JERK = max_lat_jerk
+
       # Use saturation-based torque reduction gain
       target_torque_reduction_gain = self.angle_torque_reduction_gain_controller.update(
         last_requested_angle=self.apply_angle_last,
