diff --git b/opendbc/car/hyundai/carcontroller.py a/opendbc/car/hyundai/carcontroller.py
index 8fc563b1..bcf9e1ac 100644
--- b/opendbc/car/hyundai/carcontroller.py
+++ a/opendbc/car/hyundai/carcontroller.py
@@ -174,6 +174,8 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
       self.params.ANGLE_TORQUE_OVERRIDE_CYCLES = int(self._params.get("HkgTuningOverridingCycles") or self.params.ANGLE_TORQUE_OVERRIDE_CYCLES)
       self.angle_enable_smoothing_factor = self._params.get_bool("EnableHkgTuningAngleSmoothingFactor")
 
+      self.params.ANGLE_LIMITS.MAX_ANGLE_RATE = int(self._params.get("HkgTuningAngleMaxAngleRate", return_default=True))
+
     self.angle_torque_reduction_gain_controller = TorqueReductionGainController(
       angle_threshold=.3,
       debounce_time=.1,
