diff --git b/opendbc/car/hyundai/carcontroller.py a/opendbc/car/hyundai/carcontroller.py
index 8fc563b1..93ccc6a1 100644
--- b/opendbc/car/hyundai/carcontroller.py
+++ a/opendbc/car/hyundai/carcontroller.py
@@ -160,6 +160,16 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
     # For future parametrization / tuning
     self.angle_enable_smoothing_factor = True
 
+    self.default_max_lateral_accel = self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL
+    self.default_max_lateral_jerk = self.params.ANGLE_LIMITS.MAX_LATERAL_JERK
+    self.default_max_angle_rate = self.params.ANGLE_LIMITS.MAX_ANGLE_RATE
+
+    self.custom_max_lateral_accel = self.default_max_lateral_accel
+    self.custom_max_lateral_jerk = self.default_max_lateral_jerk
+    self.custom_max_angle_rate = self.default_max_angle_rate
+    self.custom_limit_speed_kph = 32.0
+    self.custom_limit_speed_mps = self.custom_limit_speed_kph * CV.KPH_TO_MS
+
     self._params = Params() if PARAMS_AVAILABLE else None
     if PARAMS_AVAILABLE:
       self.params.ANGLE_MIN_TORQUE_REDUCTION_GAIN = parse_tq_rdc_gain(
@@ -174,6 +184,12 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
       self.params.ANGLE_TORQUE_OVERRIDE_CYCLES = int(self._params.get("HkgTuningOverridingCycles") or self.params.ANGLE_TORQUE_OVERRIDE_CYCLES)
       self.angle_enable_smoothing_factor = self._params.get_bool("EnableHkgTuningAngleSmoothingFactor")
 
+      self.custom_max_angle_rate = int(self._params.get("HkgTuningAngleMaxAngleRate", return_default=True))
+      self.custom_max_lateral_accel = float(self._params.get("HkgTuningAngleMaxLateralAccel", return_default=True))
+      self.custom_max_lateral_jerk = float(self._params.get("HkgTuningAngleMaxLateralJerk", return_default=True))
+      self.custom_limit_speed_kph = float(self._params.get("HkgTuningAngleCustomLimitMaxSpeedKph", return_default=True))
+      self.custom_limit_speed_mps = self.custom_limit_speed_kph * CV.KPH_TO_MS
+
     self.angle_torque_reduction_gain_controller = TorqueReductionGainController(
       angle_threshold=.3,
       debounce_time=.1,
@@ -204,6 +220,16 @@ class CarController(CarControllerBase, EsccCarController, LeadDataCarController,
     # angle control
     else:
       v_ego_raw = CS.out.vEgoRaw
+
+      if abs(v_ego_raw) <= self.custom_limit_speed_mps:
+        self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL = self.custom_max_lateral_accel
+        self.params.ANGLE_LIMITS.MAX_LATERAL_JERK = self.custom_max_lateral_jerk
+        self.params.ANGLE_LIMITS.MAX_ANGLE_RATE = self.custom_max_angle_rate
+      else:
+        self.params.ANGLE_LIMITS.MAX_LATERAL_ACCEL = self.default_max_lateral_accel
+        self.params.ANGLE_LIMITS.MAX_LATERAL_JERK = self.default_max_lateral_jerk
+        self.params.ANGLE_LIMITS.MAX_ANGLE_RATE = self.default_max_angle_rate
+
       desired_angle = np.clip(actuators.steeringAngleDeg, -self.params.ANGLE_LIMITS.STEER_ANGLE_MAX, self.params.ANGLE_LIMITS.STEER_ANGLE_MAX)
 
       if self.angle_enable_smoothing_factor and abs(v_ego_raw) < CarControllerParams.SMOOTHING_ANGLE_MAX_VEGO:
